### PR TITLE
Feat/reward program link

### PR DESCRIPTION
**Pull request by Space-zee**
https://app.clickup.com/t/860q7er5g
https://app.clickup.com/t/860q6fzyx

### CODE REVIEW

File: frontend/app/features/assets/controllers/AssetsDetailsController.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ import * as investAssetModel from "../models/investAssetModel";
 import {useAtom, useAtomValue, useSetAtom} from "jotai";
 import React, {useEffect, useState} from "react";
 import clsx from "clsx";
-import {$usdtBalance} from "../../../shared/usdtToken/smartContractsFactory";
+import {$doLoadUsdtBalance, $usdtBalance} from "../../../shared/usdtToken/smartContractsFactory";
```
Comment (ruslanpalagin) [inline frontend/app/features/assets/controllers/AssetsDetailsController.tsx:10]: if this is shared and required on many pages... 
could we load it on _app level?

File: frontend/app/features/assets/controllers/AssetsDetailsController.tsx [hunk 2]
```diff
@@ -111,6 +111,7 @@ console.log('$asset', $asset)
     const txId = await onFormSubmit();
     setTimeout(async () => {
       await doLoadUserTestnetProgress();
+      await doLoadUsdtBalance();
     }, 2000)
```
Comment (ruslanpalagin) [inline frontend/app/features/assets/controllers/AssetsDetailsController.tsx:115]: why do you need a timeout here?
generally this is a very bad practive...

File: frontend/app/features/mint/controllers/MintUsdtUserController.tsx [hunk 3]
```diff
@@ -52,14 +53,15 @@ export const MintUsdtUserController = () => {
...
+  if (!activeChainConfig || isUserMinted === null || balance === null) {
     return <NanoLoader className={s.loader} isLoading={isUserMinted === null || !activeChainConfig}/>
   }
 
   const onMint = async () => {
     await doMint();
     setTimeout(async () => {
       await doLoadUserTestnetProgress();
+      await doLoadUsdtBalance()
     }, 2000)
```
Comment (ruslanpalagin) [inline frontend/app/features/mint/controllers/MintUsdtUserController.tsx:65]: again - why timeout?

File: frontend/app/features/mint/controllers/MintUsdtUserController.tsx [hunk 4]
```diff
@@ -52,14 +53,15 @@ export const MintUsdtUserController = () => {
...
-  if (!activeChainConfig || isUserMinted === null) {
+  if (!activeChainConfig || isUserMinted === null || balance === null) {
     return <NanoLoader className={s.loader} isLoading={isUserMinted === null || !activeChainConfig}/>
   }
 
   const onMint = async () => {
     await doMint();
     setTimeout(async () => {
       await doLoadUserTestnetProgress();
+      await doLoadUsdtBalance()
```
Comment (ruslanpalagin) [inline frontend/app/features/mint/controllers/MintUsdtUserController.tsx:64]: if you need to reload balance - you could call this in the model that is require reloading

Comment (ruslanpalagin) [reply frontend/app/features/mint/controllers/MintUsdtUserController.tsx:64]: doMint in this case

**Review by ruslanpalagin — CHANGES_REQUESTED** [review]

File: frontend/app/features/assets/controllers/AssetsDetailsController.tsx (hunk 2 above)
Comment (Space-zee) [reply frontend/app/features/assets/controllers/AssetsDetailsController.tsx:115]: +

File: frontend/app/features/mint/controllers/MintUsdtUserController.tsx (hunk 3 above)
Comment (Space-zee) [reply frontend/app/features/mint/controllers/MintUsdtUserController.tsx:65]: +

File: frontend/app/features/assets/controllers/AssetsDetailsController.tsx (hunk 1 above)
Comment (Space-zee) [reply frontend/app/features/assets/controllers/AssetsDetailsController.tsx:10]: i think no need to load it on _app level

**Review by ruslanpalagin — APPROVED** [review]

**Pull request merged**